### PR TITLE
add obsoletes test

### PR DESCRIPTION
**Pull request by ignatenkobrain**

### CODE REVIEW

**Comment by ignatenkobrain** [comment]
@j-mracek 
```
Feature: DNF/Behave test (obsoletes test) # behave/obsoletes.feature:1

  Scenario: Install TestObsoletes from repository "test-1"                 # behave/obsoletes.feature:3
    Given I use the repository "test-1"                                    # behave/steps/test_behave.py:63
    When I execute "dnf" command "-y install TestObsoletes" with "success" # behave/steps/test_behave.py:82
    Then transaction changes are as follows                                # behave/steps/test_behave.py:136
      | State     | Packages      |
      | installed | TestObsoletes |

  Scenario: Update packages to get TestObsoletes obsoleted by TestObsoletes-ng  # behave/obsoletes.feature:10
    Given I use the repository "upgrade_1"                                      # behave/steps/test_behave.py:63
    When I execute "dnf" command "-y update" with "success"                     # behave/steps/test_behave.py:82
    Then transaction changes are as follows                                     # behave/steps/test_behave.py:136
      | State     | Packages         |
      | installed | TestObsoletes-ng |
      | removed   | TestObsoletes    |

  Scenario: Autoremove packages from repository "upgrade_1"     # behave/obsoletes.feature:18
    Given I use the repository "upgrade_1"                      # behave/steps/test_behave.py:63
    When I execute "dnf" command "-y autoremove" with "success" # behave/steps/test_behave.py:82
    Then transaction changes are as follows                     # behave/steps/test_behave.py:136
      | State   | Packages         |
      | present | TestObsoletes-ng |
      Assertion Failed: The number of found packages 0 (packages - ) for string "TestObsoletes-ng" differs from expected 1

```

Any ideas?

**Comment by ignatenkobrain** [comment]
```
Feature: DNF/Behave test (obsoletes test) # behave/obsoletes.feature:1

  Scenario: Install TestObsoletes from repository "test-1"                 # behave/obsoletes.feature:3
    Given I use the repository "test-1"                                    # behave/steps/test_behave.py:63
    When I execute "dnf" command "-y install TestObsoletes" with "success" # behave/steps/test_behave.py:82
    Then transaction changes are as follows                                # behave/steps/test_behave.py:136
      | State     | Packages      |
      | installed | TestObsoletes |

  Scenario: Update packages from "upgrade_1" to get TestObsoletes obsoleted by TestObsoletes-ng  # behave/obsoletes.feature:10
    Given I use the repository "upgrade_1"                                                       # behave/steps/test_behave.py:63
    When I execute "dnf" command "-y update" with "success"                                      # behave/steps/test_behave.py:82
    Then transaction changes are as follows                                                      # behave/steps/test_behave.py:136
      | State     | Packages         |
      | installed | TestObsoletes-ng |
      | removed   | TestObsoletes    |

  Scenario: Autoremove packages                                            # behave/obsoletes.feature:18
    When I execute "dnf" command "-y autoremove" with "success"            # behave/steps/test_behave.py:82
    When I execute "bash" command "rpm -q TestObsoletes-ng" with "success" # behave/steps/test_behave.py:82
      Assertion Failed: Return code was 1, but expected 0

    Then line from "stdout" should "start" with "TestObsoles-ng"           # None
```

Now I totally don't understand what's going on.

**Comment by j-mracek** [comment]
I create new PR, that could help with debuging

**Comment by j-mracek** [comment]
@ignatenkobrain I create new PR, that could help with debuging

**Comment by j-mracek** [comment]
@ignatenkobrain 
    Then transaction changes are as follows                     # behave/steps/test_behave.py:136
      | State   | Packages         |
      | present | TestObsoletes-ng |
The problem is that previous command maybe removed TestObsoletes-ng. Open docker, run a scenario and see et which state it fails. And which packages remain and so on.

**Comment by j-mracek** [comment]
@ignatenkobrain  Probably try to use old branch with local-build function to debug new test. Also in one feature file add only one scenario.

File: dnf-docker-test/features/obsoletes.feature [hunk 1]
```diff
@@ -0,0 +1,14 @@
+Feature: Obsoleting packages one-to-one
+
+Scenario: Updating should replace one package with another
+ Given I use the repository "test-1"
+ When I execute "dnf" command "-y install TestObsoletes" with "success"
+ Then transaction changes are as follows
+   | State     | Packages      |
+   | installed | TestObsoletes |
+ Given I use the repository "upgrade_1"
```
Comment (j-mracek) [inline dnf-docker-test/features/obsoletes.feature]: Please add test step that package TestObsoletes is user installed

File: dnf-docker-test/features/obsoletes.feature [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+ When I execute "dnf" command "-y update" with "success"
+ Then transaction changes are as follows
+   | State     | Packages         |
+   | installed | TestObsoletes-ng |
+   | removed   | TestObsoletes    |
```
Comment (j-mracek) [inline dnf-docker-test/features/obsoletes.feature]: Please add test step that TestObsoletes-ng is still userinstalled

Comment (ignatenkobrain) [inline dnf-docker-test/features/obsoletes.feature]: if it would work.... ;)

**Pull request closed without merging**